### PR TITLE
GPIO LL implementation

### PR DESCRIPTION
**Pull request by msemegen**
Fully GPIO LL implementation

### CODE REVIEW

File: peripherals/GPIO/gpio_ll.hpp [hunk 1]
```diff
@@ -9,24 +9,269 @@
...
+    };
+};
+struct gpio_odr_descriptor : private xmcu::non_constructible
+{
+    static constexpr std::uint32_t mask = 0x1u;
+    static constexpr std::uint32_t shift_multiplier = 1u;
+
+    enum class Flag : std::uint32_t
+    {
+        low,
```
[the remark this answers is not in the document]
Comment (kaczirosan) [reply peripherals/GPIO/gpio_ll.hpp:105]: copy from BSRR?

[the remark this answers is not in the document]
Comment (msemegen) [reply peripherals/GPIO/gpio_ll.hpp:105]: different registers, different values

[the remark this answers is not in the document]
Comment (msemegen) [reply peripherals/GPIO/gpio_ll.hpp]: done

File: peripherals/GPIO/gpio_ll.hpp [hunk 2]
```diff
@@ -9,24 +9,269 @@
...
 
 struct gpio : public gpio_base
 {
+private:
+    /*
+     * W - write, R - read, C - clear
+     */
+    template<typename desc_t> struct Reg_wrc
+    {
+        enum class Data;
```
[the remark this answers is not in the document]
Comment (msemegen) [reply peripherals/GPIO/gpio_ll.hpp]: done

**Review by kaczirosan — APPROVED** [review]

**Pull request merged**